readme.md

```diff
@@ -219,27 +230,28 @@ Then, go to [http://localhost:5000/](http://localhost:5000) and you should see a
   const { generate: generateId } = require('shortid')
   ```
 
-* **Your Answer:**
+* **Your Answer:** short id is a unique id generator which is used to genetrate id's for each of the veggies and fruits that are created via post. This helps to uniquely identify each of the veggie and fruits and no two id's will be the same
 
 * **Question:** What routes are currently available to you as part of this application?
 
-* **Your Answer:**
+* **Your Answer:** get /vegetables,  get /vegetables/:id, post /vegetables
 
 * **Question:** Look for `helpers.validate` in the `app.js` file. What is this and how does it work?
 
-* **Your Answer:**
+* **Your Answer:** This basically helps us to validat the post request. The body of the post should have
+a name and price as key's and thier corresponding value pairs are also expected to be passed as a part of request body and helpers.validate helps validate this
 
 * **Question:** Try creating a new vegetable. Then, try restarting your server. What happens to the data you posted and why?
 
-* **Your Answer:**
+* **Your Answer:** The data is not persisted as it is only in memory and not stored to the database. So once we restart the server that data is gone
 
 * **Question:** Take a look at the last two `app.use()` methods at the end of the file. Describe how both work and what `err` is refers to.
 
-* **Your Answer:**
+* **Your Answer:** if a particular path was not found for any of the req methods like get post put etc ,we log out the error messages using these app.use statements
 
```

Comment:
Correct, the First catches any missed requests that do not match routes, the second returns an error message.